tests/robot-tests/test_runners.py

```diff
@@ -0,0 +1,110 @@
+import os
+import reports
+import argparse
+import pabot.pabot as pabot
+import robot
+from tests.libs.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+def create_robot_arguments(arguments: argparse.Namespace, test_run_folder: str) -> []:
+    robot_args = [
+        "--name",
+        "UI Tests",
+        "--outputdir",
+        f"{test_run_folder}/",
+        "--exclude",
+        "Failing",
+        "--exclude",
+        "UnderConstruction",
+        "--exclude",
+        "VisualTesting",
+        "--xunit",
+        "xunit",
+    ]
+
+    robot_args += ["-v", f"timeout:{os.getenv('TIMEOUT')}", "-v", f"implicit_wait:{os.getenv('IMPLICIT_WAIT')}"]
+    
+    if arguments.fail_fast:
+        robot_args += ["--exitonfailure"]
+    if arguments.tags:
+        robot_args += ["--include", arguments.tags]
+    if arguments.print_keywords:
+        robot_args += ["--listener", "listeners/KeywordListener.py"]
+    if arguments.ci:
+        # NOTE(mark): Ensure secrets aren't visible in CI logs/reports
+        robot_args += ["--removekeywords", "name:operatingsystem.environment variable should be set"]
+        robot_args += ["--removekeywords", "name:common.user goes to url"]  # To hide basic auth credentials
+    
+    process_includes_and_excludes(robot_args, arguments)
+    
+    if arguments.visual:
+        robot_args += ["-v", "headless:0"]
+    else:
+        robot_args += ["-v", "headless:1"]
+    if os.getenv("RELEASE_COMPLETE_WAIT"):
+        robot_args += ["-v", f"release_complete_wait:{os.getenv('RELEASE_COMPLETE_WAIT')}"]
+    if arguments.prompt_to_continue:
+        robot_args += ["-v", "prompt_to_continue_on_failure:1"]
+    if arguments.debug:
+        robot_args += ["--loglevel", "DEBUG"]
+    robot_args += ["-v", "browser:" + arguments.browser]
+    # We want to add arguments on the first rerun attempt, but on subsequent attempts, we just want
+    # to change rerunfailedsuites xml file we use
+    robot_args += ["--output", "output.xml"]
+
+    return robot_args
+
+
+def process_includes_and_excludes(robot_args: [], arguments: argparse.Namespace):
+    if arguments.reseed:
+        robot_args += ["--include", "SeedDataGeneration"]
```

Comment:
**question** just wondering what `SeedGeneration` is responsible for? does it generate test data for test suites etc. ?

Reply:
Why hello there @luke-h1 !

It does indeed.  Lets us generate consistent seed data in each environment, and regenerate as needed.  It's a manual job currently to reseed an environment as we don't need it often.